CMakeModules/build_CLBlast.cmake

```diff
@@ -37,7 +37,7 @@ index 9446499..786f7db 100644
   set(CLBLAST_PATCH_COMMAND ${GIT} apply ${ArrayFire_BINARY_DIR}/clblast.patch)
 endif()
 
-if(WIN32 AND NOT CMAKE_GENERATOR MATCHES "Ninja")
+if(WIN32 AND CMAKE_GENERATOR_PLATFORM AND NOT CMAKE_GENERATOR MATCHES "Ninja")
```

Comment:
Interestingly, [CMAKE_GENERATOR_PLATFORM](https://cmake.org/cmake/help/latest/variable/CMAKE_GENERATOR_PLATFORM.html) is available only when user sets it. I didn't realize that earlier, I was under the impression this is auto-populated (when not explicitly provided for relevant generators) with project call/during compiler check.

Starting with CMake 3.14, there is a new variable [CMAKE_VS_PLATFORM_NAME_DEFAULT](https://cmake.org/cmake/help/v3.16/variable/CMAKE_VS_PLATFORM_NAME_DEFAULT.html#variable:CMAKE_VS_PLATFORM_NAME_DEFAULT) that contains default platform for Visual Studio generators. This variable defaults to Win32 for VS 2017 or below.